src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs

```diff
@@ -38,20 +38,11 @@ internal override bool BoolEvaluate(ConditionEvaluator.IConditionEvaluationState
                 // Check we only have one argument
                 VerifyArgumentCount(1, state);
 
-                // Expand properties and items, and verify the result is an appropriate scalar
-                string expandedValue = ExpandArgumentForScalarParameter("exists", (GenericExpressionNode)_arguments[0], state);
-
-                if (String.IsNullOrEmpty(expandedValue))
-                {
-                    return false;
-                }
-
                 try
                 {
-                    if (state.EvaluationDirectory != null && !Path.IsPathRooted(expandedValue))
-                    {
-                        expandedValue = Path.GetFullPath(Path.Combine(state.EvaluationDirectory, expandedValue));
-                    }
+                    return !ExpandArgumentAsFileList((GenericExpressionNode) _arguments[0], state)
+                        .DefaultIfEmpty()
+                        .Any(i => i == null || state.LoadedProjectsCache?.TryGet(i) == null && !FileUtilities.FileOrDirectoryExistsNoThrow(i));
```

Comment:
However much it hurts me to say this, I'd advise against Linq and variable capturing lambdas inside evaluation code. :(
This is not the hottest of paths, but it's still evaluation, which runs a lot. RPS might hiccup here.